h2o-core/src/main/java/water/parser/PreviewParseWriter.java

```diff
@@ -49,9 +62,9 @@ private void setColumnCount(int n) {
       _ndates = new int[n];
       _nnums = new int[n];
       _nempty = new int[n];
-      _domains = new IcedHashMap[n];
+      _domains = new Collection[n];
       for(int i = 0; i < n; ++i)
-        _domains[i] = new IcedHashMap<>();
+        _domains[i] = Collections.newSetFromMap(new IcedHashMap<>());
```

Comment:
you are saying that the goal of this PR is to remove the use of IcedHashMap, but yet it it still being used, I dotn get the benefit of moving just here

Reply:
See the comment above. Here I just need to have a HashSet. I couldn't have used `IcedHashSet` since it doesn't allow string to be a generic parameter.